pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/Worker.java

```diff
@@ -150,15 +150,8 @@ private static URI initialize(WorkerConfig workerConfig)
             admin.close();
         }
 
-        // initialize the dlog namespace
-        // TODO: move this as part of pulsar cluster initialization later
-        try {
-            return WorkerUtils.initializeDlogNamespace(internalConf);
-        } catch (IOException ioe) {
-            log.error("Failed to initialize dlog namespace with zookeeper {} at metadata service uri {} for storing function packages",
-                internalConf.getZookeeperServers(), internalConf.getBookkeeperMetadataServiceUri(), ioe);
-            throw ioe;
-        }
+        // get the dlog namespace
+        return WorkerUtils.getDlogNamespaceURI(internalConf.getZookeeperServers());
```

Comment:
Don't remove the code immediately. Please add a flag in Function worker config to control whether do we need to initialize dlog namespace or not. By default, we can make it `true` (to initialize dlog namespace) for backward compatibility consideration and allow people to turn it off.

Reply:
I update my code, and I'm not sure if I'm doing the right thing, please let me know if have any problems.